hfile.c

```diff
@@ -107,12 +107,19 @@ hFILE *hfile_init(size_t struct_size, const char *mode, size_t capacity)
     hFILE *fp = (hFILE *) malloc(struct_size);
     if (fp == NULL) goto error;
 
-    if (capacity == 0) capacity = 32768;
+    const int maxcap = 128*1024;
+
+    if (capacity == 0) capacity = maxcap;
     // FIXME For now, clamp input buffer sizes so mpileup doesn't eat memory
-    if (strchr(mode, 'r') && capacity > 32768) capacity = 32768;
+    if (strchr(mode, 'r') && capacity > maxcap) capacity = maxcap;
 
+#ifdef HAVE_POSIX_MEMALIGN
+    if (posix_memalign((void **)&fp->buffer, 256, capacity) < 0)
```

Comment:
One small issue.  It seems `posix_memalign()` doesn't set the buffer pointer if it fails, so it needs to be set to `NULL` on entry to avoid problems when cleaning up.

See the example on the [Open Group posix_memalign() man page](https://pubs.opengroup.org/onlinepubs/9799919799/functions/posix_memalign.html).